packages/ra-core/src/controller/edit/useEditController.spec.tsx

```diff
@@ -751,4 +914,72 @@ describe('useEditController', () => {
             previousData: undefined,
         });
     });
+
+    it('should allow to register middlewares', async () => {
+        let saveCallback;
+        const update = jest
+            .fn()
+            .mockImplementationOnce((_, { id, data }) =>
+                Promise.resolve({ data: { id, ...data } })
+            );
+        const dataProvider = testDataProvider({
+            // @ts-ignore
+            getOne: () => Promise.resolve({ data: { id: 12 } }),
+            update,
+        });
+        const middleware: Middleware<ReturnType<typeof useUpdate>[0]> = jest.fn(
```

Comment:
You should document that type in the hook jsDoc, or we'll look for it every time